docs/articles/documentation/guides/basic-guides/obtain-client-side-info.md

```diff
@@ -0,0 +1,173 @@
+
+test('My Test', async t => {
+    const docURI = await t.eval(() => document.documentURI);
+});
+```
+
+## Import Functions to be Used as Client Function Dependencies
+
+Assume you have a JS file `utils.js` with a function you need to use as a client function dependency in your test file.
+
+**utils.js**
+
+```js
+export function getDocumentURI() {
+    return document.documentURI;
+}
+```
+
+Note that TestCafe processes test files with [Babel](https://babeljs.io) internally. To avoid issues caused by code transpiling, use the [require](https://nodejs.org/api/modules.html#modules_require_id) function instead of the `import` statement to import client function dependencies.
+
+**test.js**
+
+```js
+import { ClientFunction } from 'testcafe';
+
+const getDocumentURI = require('./utils.js').getDocumentURI;
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    const getUri = ClientFunction(() => {
+        return getDocumentURI();
+    }, { dependencies: { getDocumentURI } });
+
+    const uri = await getUri();
+
+    await t.expect(uri).eql('http://devexpress.github.io/testcafe/example/');
+});
+```
+
+## Call Client Functions from Node.js Callbacks
+
+Client functions need access to the [test controller](../../reference/test-api/testcontroller/README.md) to be executed.
+When called right from the test function, they implicitly obtain the test controller.
+
+However, if you need to call a client function from a Node.js callback that fires during the test run,
+you need to manually bind this function to the test controller.
+
+You can use the `boundTestRun` option to do this.
+
+```js
+import fs from 'fs';
+import { ClientFunction } from 'testcafe';
+
+fixture `My fixture`
+    .page `http://www.example.com/`;
+
+const getDataFromClient = ClientFunction(() => getSomeData());
+
+test('Check client data', async t => {
+    const boundGetDataFromClient = getDataFromClient.with({ boundTestRun: t });
+
+    const equal = await new Promise(resolve => {
+        fs.readFile('/home/user/tests/reference/clientData.json', (err, data) => {
+            boundGetDataFromClient().then(clientData => {
+                resolve(JSON.stringify(clientData) === data);
+            });
+        });
+    });
+
+    await t.expect(equal).ok();
+});
+```
+
+This approach only works for Node.js callbacks that fire during the test run. To ensure that the test function
+does not finish before the callback does, suspend the test until the callback fires. You can introduce a promise and synchronously wait for it to complete as shown in the example above.
+
+## Client Function Limitations
+
+* You cannot use generators or `async/await` syntax within client functions.
+
+* Client functions cannot access variables defined in the outer scope in test code.
+  >You can use arguments to pass data inside these functions, except for self-invoking functions
+  that cannot take any parameters from the outside.
```

Comment:
Change to:
You can use arguments to pass data within these functions, except for self-invoking functions that do not accept outside parameters.